clang/include/clang-c/Dependencies.h

```diff
@@ -380,6 +380,244 @@ clang_experimental_DependencyScannerWorker_getFileDependencies_v5(
     CXModuleLookupOutputCallback *MLO, unsigned Options,
     CXFileDependenciesList **Out, CXDiagnosticSet *OutDiags);
 
+/**
+ * Output of \c clang_experimental_DependencyScannerWorker_getDepGraph.
+ */
+typedef struct CXOpaqueDepGraph *CXDepGraph;
+
+/**
+ * An individual module dependency that is part of an overall compilation
+ * \c CXDepGraph.
+ */
+typedef struct CXOpaqueDepGraphModule *CXDepGraphModule;
+
+/**
+ * An individual command-line invocation that is part of an overall compilation
+ * \c CXDepGraph.
+ */
+typedef struct CXOpaqueDepGraphTUCommand *CXDepGraphTUCommand;
+
+/**
+ * Settings to use for the
+ * \c clang_experimental_DependencyScannerWorker_getDepGraph action.
+ */
+typedef struct CXOpaqueDependencyScannerWorkerScanSettings
+    *CXDependencyScannerWorkerScanSettings;
+
+/**
+ * Creates a set of settings for
+ * \c clang_experimental_DependencyScannerWorker_getDepGraph action.
+ * Must be disposed with
+ * \c clang_experimental_DependencyScannerWorkerScanSettings_dispose.
+ * The provided strings are not copied; make sure the memory, that the pointer
+ * parameters point to, is still valid until the call to
```

Comment:
Nit: extra commas around "that the pointer parameters point to".  Also I think you could shorten this; e.g.

Memory for settings is not copied. Any provided pointers must be valid until the call to ...